rm_reporting/controllers/party_controller.py

```diff
@@ -107,8 +118,13 @@ def get_respondent_data(respondent_ids_string) -> dict:
 
     party_engine = app.party_db.engine
     respondent_details_query = text(f"SELECT * FROM partysvc.respondent r WHERE r.id IN ({respondent_ids_string}) ")
-    with party_engine.begin() as conn:
-        respondent_details_result = conn.execute(respondent_details_query).all()
-    results_dict = {str(getattr(item, "id")): item for item in respondent_details_result}
+    try:
+        with party_engine.begin() as conn:
+            respondent_details_result = conn.execute(respondent_details_query).all()
+        results_dict = {str(getattr(item, "id")): item for item in respondent_details_result}
+    except SQLAlchemyError as e:
+        e.statement
+        logger.error("Failed to get respondent data")
```

Comment:
As above

Reply:
I have decided to refrain from including info about the error in the logging statement, as doing so results in the SQL query getting printed in the logs.